homeassistant/components/conversation.py

```diff
@@ -153,12 +167,15 @@ class ConversationProcessView(http.HomeAssistantView):
     @RequestDataValidator(vol.Schema({
         vol.Required('text'): str,
     }))
-    @asyncio.coroutine
-    def post(self, request, data):
+    async def post(self, request, data):
         """Send a request for processing."""
         hass = request.app['hass']
 
-        intent_result = yield from _process(hass, data['text'])
+        try:
+            intent_result = await _process(hass, data['text'])
+        except intent.IntentHandleError as err:
+            intent_result = intent.IntentResponse()
+            intent_result.async_set_speech(str(err))
```

Comment:
Speaking out error messages could lead to some funny results (if the error strings are verbose) :) - but not really a problem right now anyway because there's only really the [`Error handling {}`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/intent.py#L70) string.

Reply:
`IntentHandleError` errors are meant to be spoken.

Reply:
We will have to update this btw when we start translating our intents.